modules/rostests/apitests/user32/GetDCEx.c

```diff
@@ -466,16 +463,746 @@ Test_GetDCEx_CS_SwitchedStyle()
+    ok(!dc, "Got a non-NULL DC (%p) for a destroyed window\n", dc);
+
+    dc = GetDC(hwnd_owndc);
+    ok(dc != 0, "Got NULL DC\n");
+    rop = GetROP2(dc);
+    ok(rop == R2_WHITE, "wrong ROP2 %d\n", rop);
+    ok(WindowFromDC(dc) == hwnd_owndc, "wrong window\n");
+    DestroyWindow(hwnd_owndc);
+
+    rop = GetROP2(dc);
+    ok(rop == 0, "wrong ROP2 %d\n", rop);
+    ok(WindowFromDC(dc) == 0, "wrong window\n");
+    ok(!ReleaseDC(hwnd_owndc, dc), "ReleaseDC succeeded\n");
+    dc = GetDC(hwnd_owndc);
+    ok(!dc, "Got a non-NULL DC (%p) for a destroyed window\n", dc);
+
+    DestroyWindow(hwnd_parent);
+}
+
+void Test_from_wine(void)
+{
+    WNDCLASSW cls;
+
+    cls.style = CS_DBLCLKS;
+    cls.lpfnWndProc = DefWindowProcW;
+    cls.cbClsExtra = 0;
+    cls.cbWndExtra = 0;
+    cls.hInstance = GetModuleHandleW(0);
+    cls.hIcon = 0;
+    cls.hCursor = LoadCursorW(0, MAKEINTRESOURCEW(32512));
+    cls.hbrBackground = GetStockObject(WHITE_BRUSH);
+    cls.lpszMenuName = NULL;
+    cls.lpszClassName = L"cache_class";
+    RegisterClassW(&cls);
+    cls.style = CS_DBLCLKS | CS_OWNDC;
+    cls.lpszClassName = L"owndc_class";
+    RegisterClassW(&cls);
+    cls.style = CS_DBLCLKS | CS_CLASSDC;
+    cls.lpszClassName = L"classdc_class";
+    RegisterClassW(&cls);
+    cls.style = CS_PARENTDC;
+    cls.lpszClassName = L"parentdc_class";
+    RegisterClassW(&cls);
+
+    hwnd_cache = CreateWindowW(L"cache_class", NULL, WS_OVERLAPPED | WS_VISIBLE,
+                               0, 0, 100, 100,
+                               0, 0, GetModuleHandleW(0), NULL);
+    hwnd_owndc = CreateWindowW(L"owndc_class", NULL, WS_OVERLAPPED | WS_VISIBLE,
+                               0, 200, 100, 100,
+                               0, 0, GetModuleHandleW(0), NULL);
+    hwnd_classdc = CreateWindowW(L"classdc_class", NULL, WS_OVERLAPPED | WS_VISIBLE,
+                                 200, 0, 100, 100,
+                                 0, 0, GetModuleHandleW(0), NULL);
+    hwnd_classdc2 = CreateWindowW(L"classdc_class", NULL, WS_OVERLAPPED | WS_VISIBLE,
+                                  200, 200, 100, 100,
+                                  0, 0, GetModuleHandleW(0), NULL);
+    hwnd_parent = CreateWindowW(L"static", NULL, WS_OVERLAPPED | WS_VISIBLE,
+                                400, 0, 100, 100, 0, 0, 0, NULL);
+    hwnd_parentdc = CreateWindowW(L"parentdc_class", NULL, WS_CHILD | WS_VISIBLE,
+                                  0, 0, 1, 1, hwnd_parent, 0, 0, NULL);
+
+    test_dc_attributes();
+    test_parameters();
+    test_dc_visrgn();
+    test_begin_paint();
+    test_scroll_window();
+    test_invisible_create();
+    test_dc_layout();
+    /* this should be last */
+    test_destroyed_window();
+}
+
 START_TEST(GetDCEx)
 {
-    Test_GetDCEx_Params();
     Test_GetDCEx_Cached();
     Test_GetDCEx_CS_OWNDC();
     Test_GetDCEx_CS_CLASSDC();
     Test_GetDCEx_CS_Mixed();
     Test_GetDCEx_CS_SwitchedStyle();
+    Test_from_wine();
 }
```

Comment:
Question: why "test from wine"? Don't we have them already in the winetests? (If not this means that perhaps a wine-sync of these would be in order...)